cargo-insta/tests/functional/workspace.rs

```diff
@@ -465,6 +490,138 @@ fn test_snapshot() {
     .success());
 }
 
+// Can't get the test binary discovery to work on Windows, don't have a windows
+// machine to hand, others are welcome to fix it. (No specific reason to think
+// that insta doesn't work on windows, just that the test doesn't work.)
+#[cfg(not(target_os = "windows"))]
+#[test]
+fn test_insta_workspace_root_compile_time() {
+    use std::process::Command;
+
+    use crate::TestProject;
+
+    let test_project = TestFiles::new()
+        .add_cargo_toml("insta_workspace_root_test")
+        .add_file(
+            "src/lib.rs",
+            r#"
+use insta::assert_snapshot;
+
+#[test]
+fn test_snapshot() {
+    assert_snapshot!("Hello, world!");
+}
+    "#
+            .to_string(),
+        )
+        .create_project();
+
+    let moved_workspace_compile =
+        tempfile::tempdir().expect("Unable to create temporary test directory");
+
+    // Compile test binary with INSTA_WORKSPACE_ROOT set at compile time
+    let mut cargo_cmd = Command::new("cargo");
+    TestProject::clean_env(&mut cargo_cmd);
+    let output = cargo_cmd
+        .args(["test", "--no-run"])
+        .env("INSTA_WORKSPACE_ROOT", moved_workspace_compile.path())
+        .current_dir(&test_project.workspace_dir)
+        .output()
+        .unwrap();
+    assert!(&output.status.success());
+
+    let test_binary_path = find_test_binary(&test_project.workspace_dir);
+
+    let config = rust_strings::FileConfig::new(&test_binary_path).with_min_length(5);
+    let extracted_strings =
+        rust_strings::strings(&config).expect("Unable to extract strings from binary");
+
+    assert_eq!(
+        extracted_strings
+            .iter()
+            .filter(|(s, _)| s.contains(test_project.workspace_dir.to_str().unwrap()))
+            .count(),
+        1,
+        "The final binary contains not the expected number of references to the cargo workspace"
```

Comment:
Either here or in a comment, can we answer why this should be 1? Same in the next statement

Reply:
I no strip the binaries when building so that it doesn't contain the debug symbols.

This way the test becomes more clear as I don't need to account for the additional reference in the debug symbols.

I also made the message of the assert more explicit of what is actually expected.